src/main/java/com/openshift/internal/restclient/okhttp/OpenShiftAuthenticator.java

```diff
@@ -118,27 +130,24 @@ private IAuthorizationDetails captureAuthDetails(String url) {
         return details;
     }
 
-    private String extractAndSetAuthContextToken(Response response) {
+    private String getToken(Response response) {
         String token = null;
         Map<String, String> pairs = URIUtils.splitFragment(response.header(PROPERTY_LOCATION));
         if (pairs.containsKey(ACCESS_TOKEN)) {
             token = pairs.get(ACCESS_TOKEN);
-            IAuthorizationContext authContext = client.getAuthorizationContext();
-            if (authContext != null) {
-                authContext.setToken(token);
-            }
         }
         return token;
     }
 
-    public void setOkClient(OkHttpClient okClient) {
-        this.okClient = okClient;
+    private void setToken(String token, IAuthorizationContext authorizationContext) {
+        if (authorizationContext != null) {
+            authorizationContext.setToken(token);
+        }
     }
 
-    public void setClient(DefaultClient client) {
+    public void setClient(IClient client) {
         this.client = client;
         challangeHandlers.clear();
         challangeHandlers.add(new BasicChallangeHandler(client.getAuthorizationContext()));
```

Comment:
Because you renamed Challange to Challenge, worth renaming BasicChallangeHandler to BasicChallengeHandler as well ?